### PR TITLE
Added HPA lightweight e2e test

### LINKED ISSUE #22363: Add more e2e tests for HPA
* lightweight test which can be run in gce suite
* test which verifies that version v1 works

### PR DESCRIPTION
**Pull request by piosz**
fix #22363

### CODE REVIEW

File: test/e2e/horizontal_pod_autoscaling.go [hunk 1]
```diff
@@ -61,6 +63,34 @@ var _ = Describe("Horizontal pod autoscaling (scale resource: CPU) [Serial] [Slo
 			scaleDown("rc", kindRC, rc, f)
 		})
 	})
+
+	Describe("ReplicationController light", func() {
+		It("Should scale from 1 pod to 2 pods", func() {
+			scaleTest := &HPAScaleTest{
+				initPods:   1,
+				cpuStart:   150,
```
Comment (fgrzadkowski) [inline test/e2e/horizontal_pod_autoscaling.go]: How about we change field names? I suggest 'totalInitialCPUUsage', 'perPodCPURequest' and 'targetCPUUtilizationPercent' (I'm open to different names). Each time I read these tests I'm confused with the names.

Comment (piosz) [inline test/e2e/horizontal_pod_autoscaling.go]: Done

**Comment by piosz** [comment]
PTAL

**Comment by fgrzadkowski** [comment]
Increasing priority of this change as with the planned changes around scale subresource (#21966, #21910, #22511) we need to make sure that we don't break autoscaling.

**Comment by ihmccreery** [comment]
Any reason you still do the heavyweight ones with v1beta1? It would be nice to get all of these moved to v1, would it not?

**Comment by piosz** [comment]
The plan for graduating HPA to GA is:
- for 1.2: add support for HPA in version v1
- for 1.3: migrate all core code to use new objects; depracate HPA in version v1beta1
- for 1.4: get rid off objects in v1beta1

For 1.2 all e2e tests use HPA in old version v1beta1 according to the plan (except the one I created in this PR).

**Pull request merged**